t/e2e/services/debug_api_test.go

```diff
@@ -0,0 +1,152 @@
+	s := new(DebugAPISuite)
+	s.upstream = true
+	suite.Run(t, s)
+}
+
+type DebugAPISuite struct {
+	BaseJSONRPCSuite
+	upstream bool
+}
+
+func (s *DebugAPISuite) TestAccessibleDebugAPIsUnexported() {
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, false)
+	s.NoError(err)
+	// Debug APIs should be unavailable
+	s.AssertAPIMethodUnexported("debug_postSync")
+	err = s.Backend.StopNode()
+	s.NoError(err)
+
+	err = s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+	// Debug APIs should be available
+	s.AssertAPIMethodExported("debug_postSync")
+}
+
+func (s *DebugAPISuite) TestDebugPostSyncSuccess() {
+	// Test upstream if that's not StatusChain
+	if s.upstream && GetNetworkID() == params.StatusChainNetworkID {
+		s.T().Skip()
+		return
+	}
+
+	err := s.SetupTest(s.upstream, false, true)
+	s.NoError(err)
+	defer func() {
+		err := s.Backend.StopNode()
+		s.NoError(err)
+	}()
+
+	s.addPeerToCurrentNode()
+	symID := s.generateSymKey()
+	result := s.sendPostConfirmMessage(symID)
+
+	var r struct {
+		Error struct {
+			Message string `json:"message"`
+		} `json:"error"`
+		Result hexutil.Bytes `json:"result"`
+	}
+	s.NoError(json.Unmarshal([]byte(result), &r))
+	s.Empty(r.Error.Message)
+	s.NotEmpty(r.Result)
+}
+
+// generateSymKey generates and stores a symetric key.
+func (s *DebugAPISuite) generateSymKey() string {
+	w, err := s.Backend.StatusNode().WhisperService()
+	s.Require().NoError(err)
+	symID, err := w.GenerateSymKey()
+	s.Require().NoError(err)
+
+	resp := s.Backend.CallRPC(`{"jsonrpc":"2.0","method":"shh_addSymKey",
+			"params":["` + symID + `"],
+			"id":1}`)
+	type returnedIDResponse struct {
+		Result string
+		Error  interface{}
+	}
+	symkeyAddResp := returnedIDResponse{}
+	err = json.Unmarshal([]byte(resp), &symkeyAddResp)
+	s.NoError(err)
+
+	return symID
+}
+
+// sendPostConfirmMessage calls debug_postSync endpoint with valid
+// parameters.
+func (s *DebugAPISuite) sendPostConfirmMessage(symID string) string {
+	req := whisper.NewMessage{
+		SymKeyID:  symID,
+		PowTarget: whisper.DefaultMinimumPoW,
+		PowTime:   200,
+		Topic:     whisper.TopicType{0x01, 0x01, 0x01, 0x01},
+		Payload:   []byte("hello"),
+	}
+	body, _ := json.Marshal(req)
+
+	basicCall := fmt.Sprintf(
+		`{"jsonrpc":"2.0","method":"debug_postSync","params":[%s],"id":67}`,
+		body)
+
+	return s.Backend.CallPrivateRPC(basicCall)
+}
+
+// addPeers adds a peer to the running node
+func (s *DebugAPISuite) addPeerToCurrentNode() {
+	s.Require().NotNil(s.Backend)
+	node1 := s.Backend.StatusNode().GethNode()
+	s.NotNil(node1)
+	node2 := s.newPeer("test2").GethNode()
+	s.NotNil(node2)
+
+	node1.Server().AddPeer(node2.Server().Self())
+}
+
+// newNode creates, configures and starts a new peer.
+func (s *DebugAPISuite) newPeer(name string) *node.StatusNode {
+	dir, err := ioutil.TempDir("", "test-shhext-")
```

Comment:
You can create an in-memory geth node directly:
```go
        peer, err := gethnode.New(&gethnode.Config{
		P2P: p2p.Config{
			MaxPeers:    math.MaxInt32,
			NoDiscovery: true,
			ListenAddr:  ":0",
		},
	})
```

Reply:
If not, `dir` should be cleaned up at some point.

Reply:
As discussed on PM this is a hard approach because whisper is not registered. So I'm keeping the original implementation and clear the temp dir.